src/audio/selector.c

```diff
@@ -0,0 +1,445 @@
+ */
+static int selector_ctrl_set_cmd(struct comp_dev *dev,
+			       struct sof_ipc_ctrl_data *cdata)
+{
+	//TODO add implementation aligned with with SW assumptions
+	trace_selector("selector_ctrl_set_cmd(), "
+					"cdata->cmd = %u",
+					cdata->cmd);
+
+	return 0;
+}
+
+/**
+ * \brief Gets selector control command.
+ * \param[in,out] dev Selector base component device.
+ * \param[in,out] cdata Selector command data.
+ * \return Error code.
+ */
+static int selector_ctrl_get_cmd(struct comp_dev *dev,
+			       struct sof_ipc_ctrl_data *cdata, int size)
+{
+	//TODO add implementation aligned with with SW assumptions*/
+	cdata->cmd = SOF_CTRL_CMD_ENUM;
+	trace_selector("selector_ctrl_get_cmd():"
+						"cdata->cmd = %u",
+						cdata->index);
+
+	return 0;
+}
+
+/**
+ * \brief Used to pass standard and bespoke commands (with data) to component.
+ * \param[in,out] dev Selector base component device.
+ * \param[in] cmd Command type.
+ * \param[in,out] data Control command data.
+ * \return Error code.
+ */
+static int selector_cmd(struct comp_dev *dev, int cmd, void *data,
+		      int max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+
+	trace_selector("selector_cmd()");
+
+	switch (cmd) {
+	case COMP_CMD_SET_VALUE:
+		return selector_ctrl_set_cmd(dev, cdata);
+	case COMP_CMD_GET_VALUE:
+		return selector_ctrl_get_cmd(dev, cdata, max_data_size);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * \brief Sets component component state.
+ * \param[in,out] dev Selector base component device.
+ * \param[in] cmd Command type.
+ * \return Error code.
+ */
+static int selector_trigger(struct comp_dev *dev, int cmd)
+{
+	trace_selector("selector_trigger()");
+
+	return comp_set_state(dev, cmd);
+}
+
+
+/**
+ * \brief Copies and processes stream data.
+ * \param[in,out] dev Selector base component device.
+ * \return Error code.
+ */
+static int selector_copy(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_buffer *sink;
+	struct comp_buffer *source;
+
+	tracev_selector("selector_copy()");
+
+	/* selector components will only ever have 1 source and 1 sink buffer */
+	source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
+
+	/* make sure source component buffer has enough data available and that
+	 * the sink component buffer has enough free bytes for copy. Also
+	 * check for XRUNs
+	 */
+	if (source->avail < cd->source_period_bytes) {
+		trace_selector_error("selector_copy() error: "
+					"source component buffer"
+					" has not enough data available");
+		comp_underrun(dev, source, cd->source_period_bytes, 0);
+		return -EIO;	/* xrun */
+	}
+	if (sink->free < cd->sink_period_bytes) {
+		trace_selector_error("selector_copy() error: "
+						"sink component buffer"
+						" has not enough free bytes for copy");
+		comp_overrun(dev, sink, cd->sink_period_bytes, 0);
+		return -EIO;	/* xrun */
+	}
+
```

Comment:
You have some extra linefeeds. The pre- and post commit checker helps to find them.